autode/wrappers/keywords.py

```diff
@@ -106,36 +170,39 @@ def set_ecp(self, ecp):
 
         return None
 
+    def copy(self) -> 'KeywordsSet':
+        return deepcopy(self)
+
 
-class Keywords:
+class Keywords(ABC):
 
-    def __init__(self, keyword_list=None):
+    def __init__(self,
+                 keyword_list: Union[Sequence[str], str, None] = None):
         """
-        Read only list of keywords
+        List of keywords used in an electronic structure calculation
 
-        Keyword Arguments:
-            keyword_list (list(str)): List of keywords used in a QM calculation
+        -----------------------------------------------------------------------
+        Arguments:
+            keyword_list: Keywords
         """
-        self.keyword_list = keyword_list if keyword_list is not None else []
+
+        if isinstance(keyword_list, str):
+            self._list = [keyword_list]
+
+        else:
+            self._list = list(keyword_list) if keyword_list is not None else []
 
     def __str__(self):
-        return '_'.join([str(kw) for kw in self.keyword_list])
+        return ' '.join([str(kw) for kw in self._list])
 
+    @abstractmethod
     def __repr__(self):
         """Representation of these keywords"""
```

Comment:
Have changed repr to be abstract and return strings with the prefixes in the derived classes. Makes more sense for str to just be the things in the list(?) @josephsilcock would appreciate your thoughts before merging

Reply:
seems reasonable to me...looks like you've added a dodgy file in the last commit though

Reply:
thank you 🤦